internal/librarian/command_test.go

```diff
@@ -700,6 +700,142 @@ func TestCommitAndPush(t *testing.T) {
 	}
 }
 
+func TestCopyLibraryFiles(t *testing.T) {
+	t.Parallel()
+	setup := func(src string, files []string) {
+		for _, relPath := range files {
+			fullPath := filepath.Join(src, relPath)
+			if err := os.MkdirAll(filepath.Dir(fullPath), 0755); err != nil {
+				t.Error(err)
+			}
+
+			if _, err := os.Create(fullPath); err != nil {
+				t.Error(err)
+			}
+		}
+	}
+	for _, test := range []struct {
+		name          string
+		repoDir       string
+		outputDir     string
+		libraryID     string
+		state         *config.LibrarianState
+		filesToCreate []string
+		wantFiles     []string
+		skipFiles     []string
+		wantErr       bool
+		wantErrMsg    string
+	}{
+		{
+			name:      "copy library files",
+			repoDir:   filepath.Join(t.TempDir(), "dst"),
+			outputDir: filepath.Join(t.TempDir(), "src"),
+			libraryID: "example-library",
+			state: &config.LibrarianState{
+				Libraries: []*config.LibraryState{
+					{
+						ID: "example-library",
+						SourceRoots: []string{
+							"a/path",
+							"another/path",
+						},
+					},
+				},
+			},
+			filesToCreate: []string{
+				"a/path/example.txt",
+				"another/path/example.txt",
+				"skipped/path/example.txt",
+			},
+			wantFiles: []string{
+				"a/path/example.txt",
+				"another/path/example.txt",
+			},
+			skipFiles: []string{
+				"skipped/path/example.txt",
+			},
+		},
+		{
+			name:      "library not found",
+			repoDir:   filepath.Join(t.TempDir(), "dst"),
+			outputDir: filepath.Join(t.TempDir(), "src"),
+			libraryID: "non-existent-library",
+			state: &config.LibrarianState{
+				Libraries: []*config.LibraryState{
+					{
+						ID: "example-library",
+					},
+				},
+			},
+			wantErr:    true,
+			wantErrMsg: "not found",
+		},
+		{
+			name:      "source root not found",
```

Comment:
This name suggests that this test should return an error but it actually doesn't, is this intentional?